doc/acl/Dynamic-ACL-Design.md

```diff
@@ -0,0 +1,198 @@
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+  - [Open questions](#open-questions)
+  -[Work flow](#work-flow)
+    - [Add dynamic ACL rule](#add-dynamic-acl-rule)
+    - [Refresh TTL of existing ACL rule](#refresh-ttl-of-existing-acl-rule)
+    - [Remove ACL rule when TTL expires](#remove-acl-rule-when-ttl-expires)
+  - [Testing Requirements/Design](#testing-requirementsdesign)
+    - [Unit Test cases](#unit-test-cases)
+    - [System Test cases](#system-test-cases)
+- [Open questions](#open-questions)
+
+
+## Revision
+
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Bing Wang          | Initial version                   |
+| 0.2 |9/6/2022     | Shiyan Wang        |                                   |
+
+## Scope
+
+The scope of this document covers the design of dynamic data plane ACL, which assigns a TTL to ACL rule, and removes the expired rule. Control plane ACL is not covered in this document.
+
+## Definitions/Abbreviations 
+
+| Definitions/Abbreviation | Description                                |
+|--------------------------|--------------------------------------------|
+| ACL                      | Access Control List                        |
+
+
+## Overview 
+
+The current design of data plane ACL supports only persistent rules, that is, once the rule is applied to SONiC, it will be there until manually removed or config reload.
+
+This doc proposes an enhancement to current ACL, which add a TTL to ACL rules and the rules are deleted after TTL expires.
+
+## Requirements
+
+- ACL rules are added with a TTL value
+- ACL rules are removed when TTL expired
+- ACL rules' TTL can be refreshed
+- ACL rules with TTL are kept after reboot
+
+## Architecture Design 
+
+No SONiC architecture change is required to support dynamic ACL.
+
+## High-Level Design
+
+### db schema
+
+#### CONFIG DB
+
+A new table `DYNAMIC_ACL_RULE_TABLE` is introduced to config_db to record the ACL rules with the timestamp of creation, expiration and TTL of an ACL rule.
+The new table contains all fileds of ACL rules and new fields are added: `is_dynamic`, `ttl`, `creation_time` and `expiration_time`.
+```
+key: ACL_RULE_TABLE:table_name:seq  ; key of the rule entry in the table, seq is the order of the rules   
+                                    ; when the packet is filtered by the ACL "policy_name".   
+                                    ; A rule is always assocaited with a policy.
+
+;field          = value
+is_dynamic      = true/false  ; Boolean; Optional. Indicate the rule is a time-based ACL rule.
+creation_time   = Integer     ; timestamp when the rule is created or refreshed
+expiration_time = Integer     ; timestamp when the rule is expired
+ttl             = Integer     ; the ttl, in second
```

Comment:
ttl is runtime status, let's remove it from config

Reply:
Accepted and thanks!